src/SerialPort/SerialPort.ts

```diff
@@ -86,11 +89,31 @@ export const createSerialPort = async (
         ipcRenderer.invoke(SERIALPORT_CHANNEL.SET, path, newOptions);
     };
 
-    const error = await ipcRenderer.invoke(
-        SERIALPORT_CHANNEL.OPEN,
-        options,
-        overwriteOptions
-    );
+    const tryAndOpen = async (retryCount: number) => {
+        try {
+            return (await ipcRenderer.invoke(
+                SERIALPORT_CHANNEL.OPEN,
+                options,
+                overwriteOptions
+            )) as string;
+        } catch (error) {
+            if (
+                (error as Error).message.includes(
+                    'PORT_IS_ALREADY_BEING_OPENED'
+                ) ||
+                retryCount === 0
```

Comment:
Is this the intended logic? Should it not check if it has retries left?

```suggestion
                retryCount > 0
```